### DISCUSSION TITLE
Differrent ID on web, than in API?

### DISCUSSION THREAD

**Discussion by myron0815** [trakt/trakt-api#390] [Q&A] [answered]
An user was searching for "Metropolis" via API and got:  
https://api.trakt.tv/movies/15?extended=full

This will return  
```json
{
    "title": "Metropolis",
    "year": 1927
    "ids": {
        "imdb": "tt0017136",
        "slug": "metropolis-1927",
        "tmdb": 19,
        "trakt": 15
    },
}
```

Usually, to redirect an user to the HTML detail page on trakt, i use `https://trakt.tv/movies/<TRAKT_ID>`, which usually works (redirecting to slug-id). But in this case, a completely differrent movie is returned?!  
I have a few other examples where this is the case, so what's the deal with the web? Is it better to use the slug_id? Or why are there differrences? (Metropolis 2015 / ID 354953 works as expected)

The web search for TraktId 15 `https://trakt.tv/search/trakt?query=15` also finds Metropolis...

For some yet unknown reason, _this_ case works by adding a leading zero `https://trakt.tv/movies/015` - but i doubt that this is the way to go...

**Comment by rectifyer** [comment, marked as answer]
The API prefers IDs and the website prefers slugs as the first thing it searches. The API is easily able to get the IDs which is why it's the preference there. This section in the API docs explains it in more detail: https://trakt.docs.apiary.io/#introduction/website-media-links

So in your case, I recommend using this URL for the redirect:
`/search/trakt/:id?id_type=movie`
